pkg/loki/filter.go

```diff
@@ -172,13 +187,82 @@ func notContainsKeyLineFilter(key string) lineFilter {
 		key:       key,
 		strictKey: true,
 		not:       true,
+		moreThan:  false,
+	}
+}
+
+func moreThanRegex(sb *strings.Builder, value string) {
+	// match each number greater than specified value using regex
+	// example for 123:
+	// ( 12[3-9] | 1[3-9][0-9]+ | [2-9][0-9]+ | [1-9][0-9]{3,} )
+	//       |            |              |             |
+	//       |            |              |              ↪ match number more than 1000
+	//       |            |              |
+	//       |            |               ↪ match number from 200 to 999
+	//       |            |
+	//       |             ↪ match numbers from 130 to 199
+	//       |
+	//        ↪ match any number from 123 to 129
+
+	sb.WriteString("(")
+	for i, r := range value {
+		if i < len(value)-1 {
+			sb.WriteRune(r)
+		} else {
+			sb.WriteRune('[')
+			sb.WriteRune(r)
+			sb.WriteString("-9]")
+		}
+	}
+
+	intVal, _ := strconv.Atoi(value)
+	for i := 1; i < len(value); i++ {
+		nextMin := int((intVal / int(math.Pow10(i))) + 1)
+		nextMinStr := fmt.Sprintf("%d", nextMin)
```

Comment:
Here and below: not a big deal as this wouldn't be frequently called, but `strconv.Itoa` or `strconv.FormatInt` [are more efficient](https://dev.to/mavensingh/strconvitoa-vs-strconvformatint-vs-fmtsprintf-which-is-good-to-convert-int-to-string-in-golang-3iol) to convert ints to strings.

(PS: I've seen this commit is from the DNS PR, but since it's already lgtm you might want to change it here instead)

Reply:
https://github.com/netobserv/network-observability-console-plugin/pull/365/commits/db67051648e83653abaaa08b020300c517f2d674